okhttp-clients/src/test/java/com/palantir/conjure/java/okhttp/OkHttpClientsTest.java

```diff
@@ -769,6 +769,35 @@ public void meshProxyClientChangesTargetAndInjectHostHeader() throws Exception {
         assertThat(server.takeRequest().getHeader(HttpHeaders.HOST)).isEqualTo("foo.com");
     }
 
+    @Test
+    public void non_ioexceptions_dont_break_the_world() throws IOException {
+        server.enqueue(new MockResponse().setBody("foo"));
+
+        HostEventsSink throwingSink = new HostEventsSink() {
+            @Override
+            public void record(String serviceName, String hostname, int port, int statusCode, long micros) {
+                //empty
+                throw new IllegalStateException("I am not an IOException");
```

Comment:
@carterkozak I've jammed an IllegalStateException in here to make sure the interceptor is actually wire up - what do you think?

Reply:
lgtm as long as this test fails without CatchThrowableInterceptor!

Reply:
s/fails/hangs

Reply:
Yep, confirmed you get an exception from a dispatcher thread:

```
17:43:24.535 [remoting-okhttp-dispatcher-0] ERROR com.palantir.conjure.java.okhttp.OkHttpClients - An exception was uncaught in an execution thread. This likely left a thread blocked, and is as such a serious bug which requires debugging.
java.lang.IllegalStateException: I am not an IOException
	at com.palantir.conjure.java.okhttp.OkHttpClientsTest$5.record(OkHttpClientsTest.java:780)
	at com.palantir.conjure.java.okhttp.InstrumentedInterceptor.intercept(InstrumentedInterceptor.java:64)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at com.palantir.conjure.java.okhttp.ConcurrencyLimitingInterceptor.intercept(ConcurrencyLimitingInterceptor.java:74)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at com.palantir.conjure.java.okhttp.DispatcherTraceTerminatingInterceptor.lambda$intercept$0(DispatcherTraceTerminatingInterceptor.java:32)
	at com.palantir.tracing.AsyncTracer.withTrace(AsyncTracer.java:81)
	at com.palantir.conjure.java.okhttp.DispatcherTraceTerminatingInterceptor.intercept(DispatcherTraceTerminatingInterceptor.java:32)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:210)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at com.palantir.tracing.DeferredTracer.withTrace(DeferredTracer.java:108)
	at com.palantir.tracing.Tracers$TracingAwareCallable.call(Tracers.java:338)
	at com.palantir.tracing.WrappingExecutorService.lambda$wrapTask$0(WrappingExecutorService.java:67)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

and the test hangs.